coderd/templateversions.go

```diff
@@ -631,7 +638,53 @@ func (api *API) postTemplateVersionsByOrganization(rw http.ResponseWriter, r *ht
 	var provisionerJob database.ProvisionerJob
 	err = api.Database.InTx(func(db database.Store) error {
 		jobID := uuid.New()
+		inherits := make([]uuid.UUID, 0)
 		for _, parameterValue := range req.ParameterValues {
+			if parameterValue.CopyFromParameter != uuid.Nil {
+				inherits = append(inherits, parameterValue.CopyFromParameter)
+			}
+		}
```

Comment:
Could we just inherit by default if not provided? Then we wouldn't need `CopyFromParameter`, and I believe we could delete a buncha this!

Reply:
I wanted to go down this route first, but it becomes really challenging in the backend to do this cleanly. The missing parameters are identified in the `provisioner` in the async job. 

So the BE creates the template_version, and returns the cli. The CLI watches the job, reads the missing params error, then fills in the missing params.

To do this inheritance by default in the backend, either:
1. The provisioner daemon would need to fetch the current active template version to find the missing params during the `import_job`.
2. Some backend go routine/process would need to watch the `import_job`, identify the failure, and rerun the job with updated inherited params.

If I decide to just "always inherit all params" from the last version, then you end up with excessive params. Eg: if the new template version dropped params "a" and "b", it would still inherit these values. These excessive values just seem like a bug.

----

The implementation I came up with tracks the correct params. Deleted params from the terraform file are not carried forward. New params are prompted to the user.